### ISSUE TITLE
[Bug] 1.5 HMR & vendor splitting does not work

### ISSUE THREAD

**Issue by pavel** [parcel-bundler/parcel#807]
🐛 bug report

There's a regression in `parcel 1.5`, that causes a setup with vendor splitting to fail during HMR.
The initial build and first page load works, but performing changes on files (e.g. `index.js`) causes an error in a browser.

### 🎛 Configuration

.babelrc:
```js
{
	"presets": [
		"react",
		["env", {
			"targets": {
				"browsers": [ "last 2 versions" ]
			},
			"debug": true
		}]
	]
}
```

index.html
```html
<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		<title>React App</title>
	</head>
	<body>
		<div id="app"></div>

		<script src="./index.js"></script>
	</body>
</html>
```

index.js
```js
require("babel-polyfill");

const run = async () => {

	const { React, ReactDOM } = await import("./vendor");

	const App = () => (
		<div>Hello!</div>
	);

	ReactDOM.render(<App/>, document.getElementById("app"));
};

run()
	.catch(err => console.error(err));
```

vendor.js
```js
export const React = require("react");
export const ReactDOM = require("react-dom");
```

CLI: `./node_modules/.bin/parcel index.html`

### 🤔 Expected Behavior

The same as in `1.4.1`. Changing `index.js` should reload without errors.

### 😯 Current Behavior

Error:
```
TypeError: bundles.slice is not a function
    at loadBundles (bundle-loader.js:23)
    at bundle-loader.js:11
    at new Promise (<anonymous>)
    at LazyPromise.then (bundle-loader.js:72)
```

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.5.1
| Node             | 9.5.0
| npm         | 5.6.0
| Operating System | Linux x86_64

Affected parcel versions: `1.5.0`, `1.5.1`.

Sample project code:  
[parcel-vendor-hmr-bug.tar.gz](https://github.com/parcel-bundler/parcel/files/1724842/parcel-vendor-hmr-bug.tar.gz)

**Comment by DeMoorJasper** [comment]
Does it work with current master branch?

**Comment by pavel** [comment]
@DeMoorJasper How to use current master branch?

**Comment by pavel** [comment]
@DeMoorJasper Yes, it works as expected.

**Comment by ascoders** [comment]
I have same problem, this HMR error appeared when i just add or remove any files.

**Comment by DeMoorJasper** [comment]
This should be fixed in 1.6 @ascoders
